unit-tests/shared/src/test/require-jdk9/org/scalanative/testsuite/javalib/lang/ProcessTestOnJDK9.scala

```diff
@@ -0,0 +1,149 @@
+import org.junit.Assert._
+import org.junit.Assume._
+import org.junit.Ignore
+
+import org.scalanative.testsuite.utils.Platform, Platform._
+import scala.scalanative.junit.utils.AssumesHelper._
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object ProcessTestOnJDK9 {
+  @BeforeClass
+  def checkRuntime(): Unit = {
+    assumeTrue(isMultithreadingEnabled)
+  }
+}
+
+class ProcessTestOnJDK9 {
+  import ProcessUtils._
+
+  @Test def onExitHandles(): Unit = {
+    val proc = processSleep(0.1).start()
+    val handle = proc.toHandle()
+    val f1, f2 = handle.onExit()
+    val f3, f4 = proc.onExit()
+
+    assertNotSame("each onExit invocation should return a new instance", f1, f2)
+    assertNotSame("each onExit invocation should return a new instance", f3, f4)
+    assertEquals(
+      "All completions should refer to the same pid",
+      1,
+      Seq(f1, f2, f3, f4)
+        .map(_.get())
+        .map {
+          case handle: ProcessHandle => handle.pid()
+          case proc: Process         => proc.pid()
+        }
+        .distinct
+        .size
+    )
+
+  }
+
+  @Test def onExitTerminates(): Unit = {
+    val proc = processSleep(1).start()
+    val completion = proc.onExit()
+    assertFalse(completion.isDone())
+    assertFalse(completion.isCancelled())
+
+    completion.get()
+    assertTrue(completion.isDone())
+    assertFalse(completion.isCancelled())
+    assertFalse(proc.isAlive())
+  }
+
+  // copy from ProcessTest
+  private def processForDestruction(): Process = {
+    val proc = processForCommand("ping", "-c", "2", "-i", "10", "127.0.0.1")
+      .start()
+    proc.getInputStream().read()
+    proc
+  }
+
+  @Test def processInfo(): Unit = {
+    val proc = processForDestruction()
+    try {
+      val info = proc.toHandle().info()
+      if (executingInJVM) {
+        // Might be empty sometimes on JVM but needs to be defined in SN runtime
+        assumeTrue(s"no command on JVM - $info", info.command().isPresent())
+        assumeTrue(s"no args on JVM - $info", info.arguments().isPresent())
+        assumeTrue(
+          s"no commandLine on JVM - $info",
+          info.commandLine().isPresent()
+        )
+      }
+      assertTrue(
+        s"command: ${info.command()}",
+        // On Windows every binary is wrapped in cmd invocation
+        Seq("ping", "cmd").exists(info.command().get().contains(_))
+      )
+      // On windows first argument would be ping
+      val expectedArgs = Seq("-c", "2", "-i", "10", "127.0.0.1")
+      assertEquals(
+        expectedArgs,
+        info.arguments().get().toSeq.takeRight(expectedArgs.size)
+      )
+      assertTrue(
+        s"command line (cmd): ${info.commandLine()}",
+        info.commandLine().get().contains("ping")
+      )
+      assertTrue(
+        s"command line (args): ${info.commandLine()}",
+        info.commandLine().get().contains("-c")
+      )
+      // TODO not implemented:
```

Comment:
Picky, picky...

I am wondering why there is a "TODO" here rather than a, say,
"Scala Native does not implement `Instant` or `Duration`, so do not Test those"  ?  

"TODO" implies there is a hope of them getting done some day.

Reply:
At some point we should have these, but it's a long way to go

Reply:
At least you did not say "but it's a long time to go". I am not sure if my 
word play on `Instant` & `Duration` translates well.